src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotByRequestIdInterface.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.model.SnapshotAccessRequestResponse;
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.dataset.AssetSpecification;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.SnapshotDao;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import java.time.Instant;
+import java.util.UUID;
+
+public interface CreateSnapshotByRequestIdInterface {
+  default StepResult prepareAndCreateSnapshot(
+      FlightContext context,
+      Snapshot snapshot,
+      SnapshotRequestModel snapshotReq,
+      SnapshotBuilderService snapshotBuilderService,
+      SnapshotRequestDao snapshotRequestDao,
+      SnapshotDao snapshotDao,
+      AuthenticatedUserRequest userReq)
+      throws InterruptedException {
+    UUID accessRequestId =
+        snapshotReq.getContents().get(0).getRequestIdSpec().getSnapshotRequestId();
+    SnapshotAccessRequestResponse accessRequest = snapshotRequestDao.getById(accessRequestId);
+
+    UUID dataReleaseSnapshotId = accessRequest.getSourceSnapshotId();
+    Snapshot dataReleaseSnapshot = snapshotDao.retrieveSnapshot(dataReleaseSnapshotId);
```

Comment:
I don't think the term `data release` is a general enough term to use here. In our case it's true that the snapshot referenced for this mode will be a data release snapshot but there's nothing about this code that requires that it represent a data release. So instead I would use a more general term like `requestSourceSnapshot`.